From baa90a9bf49429f3b37cf9afac66a86ef96004b8 Mon Sep 17 00:00:00 2001
From: Juergen Borleis <jbe@pengutronix.de>
Date: Thu, 19 Nov 2015 16:38:47 +0100
Subject: [PATCH 12/15] blspec: define a partition name in order to use
 Barebox's blspec feature

This entry lets barebox automatically attach to '/dev/nand0' and mounts
the new volumes on demand.

Barebox's "linux-appendroot" feature needs a partition name to create the
kernel's "ubi.mtd=<part-name>" parameter. This change also change the
paths to be searched along when the "bootchooser" is running.

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 arch/arm/dts/imx6qdl-skov-dol63x.dtsi | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/dts/imx6qdl-skov-dol63x.dtsi b/arch/arm/dts/imx6qdl-skov-dol63x.dtsi
index 6902e3f18639..81300458f948 100644
--- a/arch/arm/dts/imx6qdl-skov-dol63x.dtsi
+++ b/arch/arm/dts/imx6qdl-skov-dol63x.dtsi
@@ -54,12 +54,12 @@
 
 		system0 { /* the node's name here must match the subnode's name in the 'state' node */
 			default_attempts = <3>;
-			boot = "/dev/nand0.ubi.system0"; /* must be in conjunction with the runtime path */
+			boot = "nand0.firmware.ubi.system0"; /* must be in conjunction with the runtime path in 'dev' */
 		};
 
 		system1 { /* the node's name here must match the subnode's name in the 'state' node */
 			default_attempts = <3>;
-			boot = "/dev/nand0.ubi.system1"; /* must be in conjunction with the runtime path */
+			boot = "nand0.firmware.ubi.system1"; /* must be in conjunction with the runtime path in 'dev' */
 		};
 	};
 
@@ -260,6 +260,11 @@
 	pinctrl-0 = <&pinctrl_gpmi_nand>;
 	nand-on-flash-bbt;
 	status = "okay";
+
+	partition@0 {
+		label = "firmware";
+		reg = <0x00000000 0x000000000>; /* keep zero sized to enable autodetection */
+	};
 };
 
 /* define the SPI based 8 MiB NOR flash layout */
-- 
2.1.4

